src/test/resources/org/alfresco/repo/search/impl/parsers/cmis_test.gunit

```diff
@@ -17,6 +17,15 @@ query:
 "SELECT cmis:name,, cmis:typeId FROM cmis:document" FAIL
 "SELECT * FROM *" FAIL
 
```

Comment:
I would add a test to right outer join and expect it to throw an error

Reply:
I have added simple test for ROJ - these pass as the syntax is correct.
The failing test for ROJ at execution time will be in the related repository commit  
I will take a look there.